src/backends/r1cs/snarkjs.rs

```diff
@@ -1,10 +1,9 @@
 use crate::backends::BackendField;
-use crate::error::Result;
```

Comment:
This is a bit awkward because it seems like the errors in this crate is intended to handle the DSL's errors, but there's no concept of a span in an IO error. Maybe a different error needs to be defined?

Reply:
hmm, interesting.

Do you think this can be a workaround if it is not related to DSL source code error? --> https://github.com/zkat/miette/blob/main/README.md#-in-application-code

Reply:
I'd imagine we might find it useful to define dedicated `Error`s for each backend eventually, since IMO it might be weird to mixup DSL errors with backend errors.

In https://github.com/zksecurity/noname/pull/105/commits/28f1fb3906725005fe7dbf0842bd460f91c1cd73 I defined a new `Error` dedicated to snarkjs, perhaps we could use this as a start - What do you think?

Reply:
That makes sense.